File: src/main/java/com/github/kindrat/cassandra/client/filter/condition/CombinerCondition.java
package com.github.kindrat.cassandra.client.filter.condition;

import com.github.kindrat.cassandra.client.filter.Operator;

import java.util.Optional;
import java.util.Set;

import static com.github.kindrat.cassandra.client.util.StringUtil.wordAtPosition;
import static java.util.Arrays.stream;

public class CombinerCondition implements StateCondition {
    @Override
    public boolean isCurrentState(String[] words, Set<String> columnNames) {
        Optional<String> secondFromHead = wordAtPosition(words, -2);
        return secondFromHead.isPresent()
                && stream(Operator.values()).anyMatch(operator -> operator.getValue().equals(secondFromHead.get()));
    }

    @Override
    public State name() {
        return State.COMBINER;
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/filter/condition/CombinerCondition.kt
package com.github.kindrat.cassandra.client.filter.condition

import com.github.kindrat.cassandra.client.filter.Operator
import com.github.kindrat.cassandra.client.util.StringUtil

class CombinerCondition : StateCondition {

    override fun isCurrentState(words: Array<String>, columnNames: Set<String>): Boolean {
        val secondFromHead = StringUtil.wordAtPosition(words, -2)
        return secondFromHead?.let { word: String -> Operator.values().any { it.value == word }} ?: false
    }

    override fun name(): StateCondition.State {
        return StateCondition.State.COMBINER
    }
}